test/geometry/liegroup/test_so3.py

```diff
@@ -125,6 +125,21 @@ def test_matrix(self, device, dtype, batch_size):
             rp_ = torch.matmul(r1, pvec)[None, :]
             self.assert_close(rp_, qp_.vec)  # p_ = R*p = q*p*q_inv
 
+    @pytest.mark.parametrize("batch_size", (1, 2, 5))
+    def test_ortho(self, device, dtype, batch_size):
+        q = Quaternion.random(batch_size)
+        q = q.to(device, dtype)
+        b_R_a = So3(q).matrix()
+        a_R_b = So3(q).inverse().matrix()
+        a_R_a = (So3(q) * So3(q).inverse()).matrix()
+
+        for i in range(batch_size):
+            self.assert_close(a_R_a[i, :, :], torch.eye(3))
+            self.assert_close(b_R_a[i, :, :].inverse() * b_R_a[i, :, :], torch.eye(3))
```

Comment:
I think its because you're using element-wise multiplication here. Try it with  @

Reply:
Definitely but also I'm not quite sure that is safe to perform operations here with unhatched data, we should add some more tests

Reply:
Aha! That did it! Thank you very much @cjpurackal! I pushed the change and I'll rename this PR to reflect that this is just some added tests without any bugs found :)

Reply:
big fan you your work btw! @stevenlovegrove :)

Reply:
haha, thanks @cjpurackal!